api/v1alpha1/clienttrafficpolicy_types.go

```diff
@@ -66,6 +66,10 @@ type ClientTrafficPolicySpec struct {
 	//
 	// +optional
 	EnableProxyProtocol *bool `json:"enableProxyProtocol,omitempty"`
+	// HTTPConnectionManager provides HTTP Connection Manager configuration on the listener.
```

Comment:
this relates to https://github.com/envoyproxy/gateway/pull/2500/files

what might make sense here is a config like
```
headers:
  xForwardedFor:
    numTrustedHops: ...
```

ptal @envoyproxy/gateway-maintainers